src/ompl/base/spaces/src/DubinsStateSpace.cpp

```diff
@@ -712,21 +712,26 @@ namespace ompl::base
 DubinsStateSpace::DubinsPath dubins(double d, double alpha, double beta)
 {
     if (d < DUBINS_EPS && fabs(alpha - beta) < DUBINS_EPS)
-        return {&DubinsStateSpace::dubinsPathType[0], 0, d, 0};
+        return {&DubinsStateSpace::dubinsPathType()[0], 0, d, 0};
     alpha = mod2pi(alpha);
     beta = mod2pi(beta);
     return isLongPath(d, alpha, beta) ? ::dubinsClassification(d, alpha, beta) : ::dubinsExhaustive(d, alpha, beta);
 }
 
-// const DubinsStateSpace::DubinsPathSegmentType DubinsStateSpace::dubinsPathType[6][3] = {
-const std::vector<std::vector<DubinsStateSpace::DubinsPathSegmentType> > DubinsStateSpace::dubinsPathType {{
-    {{DUBINS_LEFT, DUBINS_STRAIGHT, DUBINS_LEFT}},
-    {{DUBINS_RIGHT, DUBINS_STRAIGHT, DUBINS_RIGHT}},
-    {{DUBINS_RIGHT, DUBINS_STRAIGHT, DUBINS_LEFT}},
-    {{DUBINS_LEFT, DUBINS_STRAIGHT, DUBINS_RIGHT}},
-    {{DUBINS_RIGHT, DUBINS_LEFT, DUBINS_RIGHT}},
-    {{DUBINS_LEFT, DUBINS_RIGHT, DUBINS_LEFT}}
-  }};
+const std::vector<std::vector<DubinsStateSpace::DubinsPathSegmentType> >& DubinsStateSpace::dubinsPathType() {
+  static std::vector<std::vector<DubinsStateSpace::DubinsPathSegmentType> >* pathType
```

Comment:
Can't this be `std::vector<std::vector<DubinsStateSpace::DubinsPathSegmentType> >` (to avoid heap allocation) and return `pathType`?

Reply:
The idea discussed in this article (https://ppwwyyxx.com/blog/2023/Safe-Static-Initialization-No-Destruction), is that the destructor is never called, to ensure that the static storage object does not go out of scope for any other objects referencing it at library unload. Not using a pointer will result in the same issue we currently have, unless we use the [`NoDestructor<>`](https://chromium.googlesource.com/chromium/src/base/+/40343e3fbc1e4835145f125adedcf15a2eb4542f/no_destructor.h) wrapper referred to, which we could do at the cost of introducing the extra template class. I though the pointer to be the simpler of the two, but if heap is a concern I can add the template.

Reply:
This is what the `NoDestructor<>` version would look like:
https://github.com/srmainwaring/ompl/tree/prs/pr-dubins-path-static-init-v2

The `NoDestructor<>` is reproduced directly from the link above, I've included the original copyright notice - may need updating to comply with the copyright / license notice for OMPL.

Reply:
ok, I only skimmed the article you linked to the first time. I read it more carefully now. This solution makes sense. There is not a big additional benefit to the `NoDestructor<>` solution, so I'm fine with merging this.